lib/ask/runtime/flow.ex

```diff
@@ -501,6 +501,31 @@ defmodule Ask.Runtime.Flow do
       flow.questionnaire.steps
     end
   end
+
+  def interim_partial_by_relevant_steps?(flow) do
+    is_relevant_step = fn step -> step["relevant"] end
+    steps(flow) |> Enum.any?(is_relevant_step)
+  end
+
+  def min_relevant_steps(%{questionnaire: questionnaire} = _flow) do
+    questionnaire.partial_config["min_relevant_steps"]
+  end
+
+  def relevant_response?(%{questionnaire: questionnaire} = _flow, response) do
+    quiz_step = questionnaire.steps |> Enum.find(fn step -> step["store"] == response.field_name end)
+    not_ignored = fn value -> String.upcase(value) not in ignored_values_from_relevant_steps(questionnaire) end
```

Comment:
This is executing `ignored_values_from_relevant_steps` multiple times, right? Can we do it once only?

Reply:
✅ done 
now is stored if the `Flow` struct
Is calculated once: when the Flow is started or loaded from the db